src/metabase/api/embed/common.clj

```diff
@@ -125,9 +125,22 @@
   not automatically converted. Thus we must do it ourselves here to make sure things are done as we'd expect.
   Also, any param values that are blank strings should be parsed as nil, representing the absence of a value."
   [query-params]
-  (-> query-params
-      (update-keys keyword)
-      (update-vals (fn [v] (if (= v "") nil v)))))
+  (letfn [(maybe-read [v]
+            (if (string? v)
+              (cond
+                (#{"true" "false"} v) (parse-boolean v)
```

Comment:
I don't have any brilliant suggestions, but it seems crazy that this is the first time we've had to deal with this? I'm surprised there's not a lib function somewhere that replicates this entire `cond`

Reply:
@escherize are we missing a trick here with using `malli.schema.BooleanValue` and friends?

Reply:
Yea, I'd also gladly use an existing util fn or something else if there's something obvious!

Reply:
I'm not sure we should be converting every single "true" query parameter value to be `true`, shouldn't that be driven by the arg->schema map in defendpoint instead? It would seem like a surprise if I chose true or false for a collection/db/query/etc name, (that got sent through a qp) this could break when e.g. inserting `true` where we rightly expect "true".

If we do it this way we should add some docs that say you're not allowed to represent the strings true or false as query param values

Reply:
> are we missing a trick here with using malli.schema.BooleanValue and friends?

Yeah, I think we should just coerce the query param value inside of defendpoint.

Reply:
I think the mechanim to convert these doesn't matter. I was under the impression that there is a way to go from the name of the key to the expected type, and I was going to advocate for deriving the proper conversion function that way. @adam-james-v has informed me that there actually isn't a good way to know what type a value in these query params should actually be, so I think this guess is fine, and should fix the P1.

We should probably revisit this and figure out how to actually type the query param data.